skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsperiod.java

```diff
@@ -0,0 +1,79 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.common.ag114.v1.model.converter.certificate.question;
+
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_HEADER_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_PERIOD_JSON_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_PERIOD_SVAR_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_FROM_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_TOM_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_PERIOD_SVAR_TEXT_ID;
+import static se.inera.intyg.common.support.facade.util.ValidationExpressionToolkit.singleExpression;
+import static se.inera.intyg.common.support.facade.util.ValueToolkit.dateRangeValue;
+
+import se.inera.intyg.common.services.texts.CertificateTextProvider;
+import se.inera.intyg.common.support.facade.model.Certificate;
+import se.inera.intyg.common.support.facade.model.CertificateDataElement;
+import se.inera.intyg.common.support.facade.model.config.CertificateDataConfigDateRange;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidation;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidationMandatory;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueDateRange;
+import se.inera.intyg.common.support.model.InternalLocalDateInterval;
+
+public class QuestionSjukskrivningsperiod {
+
+    public static CertificateDataElement toCertificate(InternalLocalDateInterval sjukskrivningsPeriod, int index,
+        CertificateTextProvider textProvider) {
+        return CertificateDataElement.builder()
+            .id(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_ID)
+            .index(index)
+            .parent(SJUKSKRIVNINGSGRAD_HEADER_ID)
+            .config(
+                CertificateDataConfigDateRange.builder()
+                    .id(SJUKSKRIVNINGSGRAD_PERIOD_JSON_ID)
+                    .text(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_TEXT_ID))
+                    .fromLabel(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_FROM_ID))
+                    .toLabel(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_TOM_ID))
+                    .build()
+            )
+            .value(
+                CertificateDataValueDateRange.builder()
+                    .id(SJUKSKRIVNINGSGRAD_PERIOD_JSON_ID)
+                    .from(sjukskrivningsPeriod != null && sjukskrivningsPeriod.getFrom() != null ? sjukskrivningsPeriod.fromAsLocalDate()
```

Comment:
Redundant check. sjukskrivningsPeriod.fromAsLocalDate() handles if from is null, so you could simplify the statement by removing && sjukskrivningsPeriod.getFrom() != null.

Same from getTom().

Reply:
Good suggesstion! Fixed!